lib/lte_link_control/lte_lc.c

```diff
@@ -1500,9 +1501,26 @@ int lte_lc_lte_mode_get(enum lte_lc_lte_mode *mode)
 	return 0;
 }
 
-int lte_lc_neighbor_cell_measurement(void)
+int lte_lc_neighbor_cell_measurement(enum lte_lc_neighbor_search_type type)
 {
-	return at_cmd_write(AT_NCELLMEAS_START, NULL, 0, NULL);
+	int err;
+
+	/* Starting from modem fw v1.3.1, there's an optional parameter to specify
+	 * the type of search. First we try to send the new version of the command.
+	 * If it fails, we assume that it's an older version and attempts without
+	 * the parameter. This corresponds to LTE_LC_NEIGHBOR_SEARCH_TYPE_DEFAULT.
+	 */
+	err = nrf_modem_at_printf("AT%%NCELLMEAS=%d", type);
+	if (err) {
+		LOG_WRN("AT command failed, retrying with legacy version of the command");
+		LOG_WRN("Note that neighbor search type is supported for modem fw >= v1.3.1");
+		LOG_WRN("This will use LTE_LC_NEIGHBOR_SEARCH_TYPE_DEFAULT as search type");
+
+		/* Retry without parameter */
+		return nrf_modem_at_printf("AT%%NCELLMEAS");
```

Comment:
Because LTE_LC_NEIGHBOR_SEARCH_TYPE_DEFAULT is the default search type, the warning could be printed only in case the requested search type is not _DEFAULT and modem rejects the command with the requested type. Or you could even use plain AT%NCELLMEAS command if requested search type is LTE_LC_NEIGHBOR_SEARCH_TYPE_DEFAULT, which is most likely the most common use case. This would avoid unnecessary warning traces with pre-1.3.1 MFWs.